lib/game_logic.rb

```diff
@@ -0,0 +1,54 @@
+class Board
+  attr_accessor :valid, :square
+  attr_reader :game
```

Comment:
Here `:square` is not needed as attr_accessor. you should use `:square` as attr_reader.